Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs

```diff
@@ -153,97 +156,174 @@ public async Task<GetStatsResponse> GetStatsAsync(GetStatsRequest request, Cance
-                if (openStreamResult.Succeeded)
-                {
-                    await StreamContentAsync(openStreamResult, responseStream, startTime);
-                }
-                else
-                {
-                    await responseStream.WriteAsync(
-                        new CopyFileResponse
+                    // Send the content.
+                    if (result.Succeeded)
+                    {
+                        byte[] buffer = new byte[ContentStore.Grpc.CopyConstants.DefaultBufferSize];
+                        switch (compression)
                         {
-                            Header = new ResponseHeader(startTime, false, (int)openStreamResult.Code, openStreamResult.ErrorMessage, openStreamResult.Diagnostics),
-                        });
+                            case CopyCompression.None:
+                                await StreamContentAsync(result.Stream, buffer, responseStream, context.CancellationToken);
+                                break;
+                            case CopyCompression.Gzip:
+                                await StreamContentWithCompressionAsync(result.Stream, buffer, responseStream, context.CancellationToken);
+                                break;
+                        }               
+                    }
                 }
+            
             }
-            catch (Exception e)
+            catch (Exception)
             {
-                await responseStream.WriteAsync(
-                    new CopyFileResponse
-                    {
-                        Header = new ResponseHeader(startTime, false, (int)CopyFileResult.ResultCode.Unknown, e.ToString()),
-                    });
+                throw;
             }
         }
 
-        private async Task StreamContentAsync(OpenStreamResult openStreamResult, IServerStreamWriter<CopyFileResponse> responseStream, DateTime startTime)
+        private async Task<(long, long)> StreamContentAsync(Stream reader, byte[] buffer, IServerStreamWriter<CopyFileResponse> responseStream, CancellationToken cts = default(CancellationToken))
+        {
+            Debug.Assert(!(reader is null));
+            Debug.Assert(!(responseStream is null));
+
+            long chunks = 0L;
+            long bytes = 0L;
+            while (true)
+            {
+                int chunkSize = await reader.ReadAsync(buffer, 0, buffer.Length, cts).ConfigureAwait(false);
+                if (chunkSize == 0) { break; }
+
+                ByteString content = ByteString.CopyFrom(buffer, 0, chunkSize);
+                CopyFileResponse response = new CopyFileResponse() { Content = content, Index = chunks };
+                await responseStream.WriteAsync(response).ConfigureAwait(false);
+
+                bytes += chunkSize;
+                chunks++;
+
+                Console.WriteLine(chunks);
+            }
+
+            return (chunks, bytes);
+        }
+
+        /*

```

Comment:
>/* [](start = 8, length = 2)

Would you mind removing the commented out code?

Reply:
Ack, sorry, fixed.